### DISCUSSION TITLE
symfony 6.1 & mssql 2008

### DISCUSSION THREAD

**Discussion by peterludwigfft** [symfony/symfony#46975] [Q&A] [answered]
Hello out there,
I try to configure the correct "DATABASE_URL" for the 2008 MS SQLServer.
The problem is the created SQL string by doctrine. The newer "OFFSET" construct is used here instead of the "LIMIT". However, this leads to an SQL error in the 2008 SQL Server.
I have already tried different combination in the ".env" file, e.g. >Database_url = "mssql: // user: passwd@IP: 1433/bmw_lpz_test? Server version = 10.50.1600"<

Do you have any ideas?

with regards Peter

**Comment by stof** [comment, marked as answer]
See this changelog entry for doctrine/dbal 3.0: https://github.com/doctrine/dbal/blob/3.3.x/UPGRADE.md#bc-break-removed-support-for-sql-server-2008-and-older

If you want to use mssql 2008, you will need to use doctrine/dbal 2.x

**Comment by peterludwigfft** [comment]
Thank You.
So, do I have to use Symfony 5.x to get things run?

**Reply by stof** [reply]
No. you can keep using Symfony 6 if you want. We haven't dropped support for doctrine/dbal 2.x.
What you need is downgrading doctrine/dbal, not symfony.

**Reply by peterludwigfft** [reply]
Thank You very much. 
I got it now by adding a new line in the composer.json right for "doctrine/dbal": "~2",
Software is now running.

**Comment by w455im** [comment]
Hello,
I appologize for reviving an old discussion. 
How did you connect Symfony 6.1 to SQL Server 2008 ? I can't get it to connect using **ODBC Driver 18** and this in my .env file: `Database_url = "sqlsrv://user: passwd@IP: 1433/oontime"`. It keeps returning this error message 

> No connection could be made because the target machine actively refused it

**Reply by peterludwigfft** [reply]
Hi,
I use Apache with PHP. In PHP the extension "pdo_sqlsrv_81_ts_x64" is loaded.
The extension is from the "Microsoft Drivers 5.10 for PHP for SQL Server Jan 2022".
In the env-file the following command was set: >DATABASE_URL="mssql://user:password@ip:1433/databasename?serverVersion=10.50.4000.0"<
